tests/common/include/iot_config_common.h

```diff
@@ -57,7 +57,8 @@
 #endif
 
 /* Use POSIX types on FreeRTOS. */
-#include "platform/iot_platform_types_posix.h"
```

Comment:
I never really liked the fact that we have to swap out header files to swap implementations. A strong abstraction would only require us to swap out .o files, but of course if you do not have dynamic allocation and your e.g. taskPool wants to contain a mutex which is statically allocated this is necessary so I kept it like this.